### ISSUE TITLE
httpApi timeout over 29 seconds message on functions that don't use the httpApi

### ISSUE THREAD

**Issue by Nemo64** [serverless/serverless#7440]
# Bug Report

## Description

I have a function definition like this:
```yaml
functions:

  website:
    handler: public/index.php
    timeout: 15 # seconds (max would be 29 though api gateway limitation)
    layers:
      - ${bref:layer.php-73-fpm}
    events:
      - httpApi: '*'

  console:
    handler: bin/console.php
    timeout: 120 # in seconds
    layers:
      - ${bref:layer.php-73}
      - ${bref:layer.console} # The "console" layer
```

This creates a warning
 
```
 Serverless Warning --------------------------------------
 
  Function timeout setting (120) is greater than maximum allowed timeout for HTTP API endpoint (29s). This may introduce situation where endpoint times out for succesful lambda invocation.
```

However, the function with 120 seconds timeout is not registered for the httpApi.
This was introduced in #7420

**Comment by jgeurts** [comment]
Seeing the same warning for lambdas triggered by cloudfront event:

```
  deleteUnusedLambdaVersions:
    handler: src/services/public/deleteUnusedLambdaVersions.handler
    memorySize: 128
    reservedConcurrency: 1
    events:
      - schedule: rate(1 day)
    timeout: 300
```

Gives the following warning:

```
Serverless Warning --------------------------------------
Function timeout setting (300) is greater than maximum allowed timeout for HTTP API endpoint (29s). This may introduce situation where endpoint times out for succesful lambda invocation.
```

**Comment by jgeurts** [comment]
For what it's worth, we did not see that error with Serverless v1.65.0. It seems to be introduced with 1.66.0

**Comment by mhotchen** [comment]
Samesies. Every Lambda in my account triggers this warning on deployment now regardless of what it's attached to, including ones attached to an API Gateway endpoint (which is supposed to timeout after 30 seconds I believe, not 29 seconds).

No big deal everything is still running smooth as butter. It's just a little bit of log pollution for me.

If this warning were being produced for a specific Lambda I wouldn't know which one is causing the issue from the warning itself which would make it difficult to debug. It would be useful to include the name of the handler in the warning message.

**Comment by MEGApixel23** [comment]
Even if a timeout is set to 29 it gives the following warning:

```
Serverless Warning --------------------------------------

  Function timeout setting (29) is greater than maximum allowed timeout for HTTP API endpoint (29s). This may introduce situation where endpoint times out for succesful lambda invocation.
```

**Comment by rmolinamir** [comment]
Downgrading to `v1.65.00` fixed it for me, thanks @jgeurts.

**Comment by medikoo** [comment]
@Nemo64 thanks for reporting, will be fixed with #7467 

@mhotchen I've also ensured that function name is provided in eventual warning.

As you noted it's an only a log pollution issue, it doesn't affect functionality.

**Comment by mhotchen** [comment]
@medikoo amazing; thanks for the fast response!

I work at a company that has helped over 10k people quit smoking. We rely heavily on serverless. I, and the whole company, really appreciate your work.
